api/config/v1/health.go

```diff
@@ -0,0 +1,229 @@
+		}
+		return fmt.Errorf("invalid string value for criticalXIDs: %s", str)
+	}
+
+	// Try to unmarshal as array of numbers
+	var xids []uint64
+	if err := json.Unmarshal(data, &xids); err == nil {
+		c.All = false
+		c.XIDs = xids
+		return nil
+	}
+
+	return fmt.Errorf("criticalXIDs must be either \"all\" or an array of numbers")
+}
+
+// MarshalJSON implements custom JSON marshaling for CriticalXIDsType
+func (c CriticalXIDsType) MarshalJSON() ([]byte, error) {
+	if c.All {
+		return json.Marshal("all")
+	}
+	return json.Marshal(c.XIDs)
+}
+
+// UnmarshalYAML implements custom YAML unmarshaling for CriticalXIDsType
+func (c *CriticalXIDsType) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	// Try to unmarshal as string first
+	var str string
+	if err := unmarshal(&str); err == nil {
+		if strings.ToLower(str) == "all" {
+			c.All = true
+			c.XIDs = nil
+			return nil
+		}
+		return fmt.Errorf("invalid string value for criticalXIDs: %s", str)
+	}
+
+	// Try to unmarshal as array of numbers
+	var xids []uint64
+	if err := unmarshal(&xids); err == nil {
+		c.All = false
+		c.XIDs = xids
+		return nil
+	}
+
+	return fmt.Errorf("criticalXIDs must be either \"all\" or an array of numbers")
+}
+
+// MarshalYAML implements custom YAML marshaling for CriticalXIDsType
+func (c CriticalXIDsType) MarshalYAML() (interface{}, error) {
+	if c.All {
+		return "all", nil
+	}
+	return c.XIDs, nil
+}
+
+// DefaultHealth returns the default health configuration for standard deployments
+func DefaultHealth() *Health {
+	return &Health{
+		Disabled: false,
+		EventTypes: []string{
+			"EventTypeXidCriticalError",
+			"EventTypeDoubleBitEccError",
+			"EventTypeSingleBitEccError",
+		},
+		IgnoredXIDs: []uint64{
+			13,  // Graphics Engine Exception
+			31,  // GPU memory page fault
+			43,  // GPU stopped processing
+			45,  // Preemptive cleanup, due to previous errors
+			68,  // Video processor exception
+			109, // Context Switch Timeout Error
+		},
+		CriticalXIDs: &CriticalXIDsType{
+			All: true,
+		},
+	}
+}
+
+// IsCritical checks if a given XID should be treated as critical
+func (h *Health) IsCritical(xid uint64) bool {
+	// If health checks are disabled, nothing is critical
+	if h.Disabled {
+		return false
+	}
+
+	// Check if XID is in ignored list
+	for _, ignoredXID := range h.IgnoredXIDs {
+		if xid == ignoredXID {
+			return false
+		}
+	}
+
+	// If no critical XIDs specified, default to all
+	if h.CriticalXIDs == nil {
+		return true
+	}
+
+	// If all XIDs are critical (except ignored ones)
+	if h.CriticalXIDs.All {
+		return true
+	}
+
+	// Check if XID is in critical list
+	for _, criticalXID := range h.CriticalXIDs.XIDs {
+		if xid == criticalXID {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Validate checks if the health configuration is valid
+func (h *Health) Validate() error {
+	if h == nil {
+		return nil
+	}
+
+	// Validate event types
+	validEventTypes := map[string]bool{
+		"EventTypeXidCriticalError":  true,
```

Comment:
@ArangoGutierrez rn, we only look at XID and ECC errors should we also consider GPU power and temp?

Reply:
That's a good discussion, but out of scope for this PR